### PR TITLE
Benchmark comparison refactor

### PR DESCRIPTION
**Pull request by rsevilla87**
Trying to simplify the code in order to ease contributions, improvements and debugging. Pending tasks:

- [x] CSV output to file
- [x] Standard output based on python's tabulate.
- [x] Collate. Deprecated now
- [x] Replaced ES Search by Scan (replaced by aggs size)
- [x] Increment size of returned aggregations
- [x] Improve kube-burner comparison
- [x] Update docs

Additional changes:
- Code optimization
- PEP8
- Code comments
- Remove collate
- Remove `emit_compare_map` references

Signed-off-by: Raul Sevilla <rsevilla@redhat.com>

### CODE REVIEW

File: src/touchstone/compare.py [hunk 1]
```diff
@@ -197,101 +187,38 @@ def main(args):
...
-                compare_dict(compute_uuid_dict, args.identifier,
-                             compute_aggs_set, _compute_value,
-                             compute_buckets, args.uuid, _compute_header,
-                             max_level=2 * len(compute_buckets),
-                             output_file=output_file, csv=print_csv)
+                result = database_instance.emit_compute_dict(uuid=uuid,
+                                                             compute_map=compute,
+                                                             index=index,
+                                                             identifier=args.identifier)
+                for b in compute["buckets"]:
```
Comment (jtaleric) [inline src/touchstone/compare.py:208]: s/b/bucket

File: src/touchstone/databases/elasticsearch.py [hunk 2]
```diff
@@ -1,183 +1,134 @@
...
+        """
+        output_dict = {}
+        buckets = compute_map['buckets']
+        aggregations = compute_map['aggregations']
+        collate = compute_map['collate']
+        filters = compute_map['filter']
+
+        logger.debug("Initializing search object")
+        kw_identifier = identifier + ".keyword"  # append .keyword
+        s = Search(using=self._conn_object, index=str(index)).query("match", **{kw_identifier: uuid})
```
Comment (jtaleric) [inline src/touchstone/databases/elasticsearch.py:67]: We should replace Search with Scan, which is going to change the structure of the response.

**Comment by amitsagtani97** [comment]
@rsevilla87 it is already using the tabulate to generate the output

**Comment by rsevilla87** [comment]
> @rsevilla87 it is already using the tabulate to generate the output

no, only for metadata

**Comment by rsevilla87** [comment]
Example outputs csv and stdout:

https://termbin.com/24r2
https://termbin.com/23la
https://termbin.com/vvvt

**Review by amitsagtani97 — COMMENTED** [review]
We can remove the changes in tox.ini file from this pr and merge this - https://github.com/cloud-bulldozer/benchmark-comparison/pull/37 later.

**Review by jtaleric — COMMENTED** [review]
This is outstanding work @rsevilla87 ! 

I do have one question, why did this query return an odd number?

touchstone_compare kubeburner elasticsearch ripsaw -url LOLPW:LOLPW my-LOLSERVER:9200 -u a4a69580-4f90-43a4-b99b-f8dd64e3e08e 6c56bf93-6b06-47b4-acba-68b3e69d9fa4

I see only one result for : 

`rxDroppedPackets-AggregatedInfra` 

Since we are doing comparisons, should we not show incomplete data? ie if `rxDroppedPackets-AggregatedInfra` doesn't exist for both queries, don't show it?

**Comment by rsevilla87** [comment]
> This is outstanding work @rsevilla87 !
> 
> I do have one question, why did this query return an odd number?
> 
> touchstone_compare kubeburner elasticsearch ripsaw -url LOLPW:LOLPW my-LOLSERVER:9200 -u a4a69580-4f90-43a4-b99b-f8dd64e3e08e 6c56bf93-6b06-47b4-acba-68b3e69d9fa4
> 
> I see only one result for :
> 
> `rxDroppedPackets-AggregatedInfra`
> 
> Since we are doing comparisons, should we not show incomplete data? ie if `rxDroppedPackets-AggregatedInfra` doesn't exist for both queries, don't show it?

I guess one of the UUIDs you queried doesn't have that metrics. With respect to prevent showing incomplete data, makes sense, will implement that.

**Comment by rsevilla87** [comment]
Added "config file" feature, can be used with the --config flag.
```
$ touchstone_compare kubeburner elasticsearch ripsaw -url <es> -u <uuid array> -o json  --config examples/kubeburner.json 
```

**Review by mohit-sheth — APPROVED** [review]
LGTM from https://github.com/cloud-bulldozer/benchmark-comparison/pull/36#issuecomment-735451853

**Review by jtaleric — APPROVED** [review]
LGTM!

**Pull request merged**